.github/workflows/publish.yaml

```diff
@@ -0,0 +1,222 @@
+        default: ${{ github.repository }}
+        type: string
+      registry_branch:
+        description: The branch of the Bazel registry to open a PR against. Defaults to main.
+        default: main
+        type: string
+    secrets:
+      publish_token:
+        required: true
+        description: A Classic Personal Access Token (PAT) used for pushing to a registry fork and opening up a pull request.
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the module repository
+      uses: actions/checkout@v4.2.2
+      with:
+        ref: ${{ github.event.inputs.tag_name }}
+        repository: ${{ inputs.repository }}
+        path: this
+
+    - name: Checkout BCR
+      uses: actions/checkout@v4.2.2
+      with:
+        repository: ${{ github.event.inputs.registry }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+        path: bazel-central-registry
+
+    # Get version from the tag, stripping any v-prefix
+    - name: Write release version
+      env:
+        TAG: ${{ inputs.tag_name }}
+      run: |
+        VERSION=${TAG#v}
+        echo Version: $VERSION
+        echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+    # Remove any pre-existing attestations.template.json files so that the following (dummy) entry
+    # creation for generating attestations will succeed without trying to substitute and verify
+    # existing attestations. Any existing templates will be restored when the final entry is created.
+    - name: Remove attestations.template.json
+      working-directory: this/.bcr
+      run: find . -type f -name 'attestations.template.json' -delete
+
+    # Create an initial entry so that we can attest the generated source.json and MODULE.bazel
+    # files. These are needed to solve a chicken and egg problem where the attestations are referenced
+    # by attestations.template.json entry file, which is included in the entry published later on.
+    # This entry will be discarded.
+    - name: Create entry
+      id: create-entry
+      uses: bazel-contrib/publish-to-bcr@fa7d3c8ad241c2c0cde639eb2225be55816e9565
+      with:
+        attest: true
+        attestations-dest: attestations
+        tag: ${{ inputs.tag_name }}
+        module-version: ${{ env.VERSION }}
+        local-registry: bazel-central-registry
+        templates-dir: this/.bcr
+
+    # Upload the attestations to the release. This will override attestations that
+    # were already uploaded on a previous run.
+    - name: Upload attestations to release
+      uses: softprops/action-gh-release@v1
+      with:
+        files: attestations/*
+        repository: ${{ inputs.repository }}
+        tag_name: ${{ inputs.tag_name }}
+
+    # Publish to BCR can run substitutions on an attestations.template.json file. Add a default
+    # template here and don't require it to exist in the module repo's .bcr templates folder.
+    - name: Create attestations template
+      working-directory: this/.bcr
```

Comment:
Although this is a long script and it's tempting to put it in a file, it's surprisingly difficult to try to reference a file in a reusable workflow. This is because it runs in the context of the calling workflow. GitHub does not provide any path with a checkout of the reusable workflow repository, and it does not provide the ref used. The only way to properly reference a file is to pass in the reusable workflow ref/tag as an input, then manually check it out at that ref and retrieve the file.

Reply:
SGTM, maybe better as a comment in the file rather than just on the PR